File: examples/javascript/searchAppointments.js
// jQuery 3.1.0
// https://cdnjs.cloudflare.com/ajax/libs/jquery/3.1.1/jquery.min.js
// ES5

var _basePath = 'https://[yourServerName]/SRSAPI/Generic/';

function login(userName, password) {
  return $.ajax({
    url: _basePath + 'Token',
    data: {
    	userName: userName,
      password: password,
      dataSourceId:0
    },
    type: "POST",
    datatype: 'json'
  }).catch(function(ex){
    console.log(ex);
    alert('Login Failed');
  });
}

function searchAppointments(token, personId){
  return $.ajax({
    //Find all appointments by PersonId
    url: _basePath + 'Appointment?PersonId=' + personId,
    type: "GET",
    datatype: 'json'
  }).catch(function(ex){
    console.log(ex);
    alert('Request Failed');
  });
}

login('yourUserName', 'yourPassword')
.then(function(token){
    return searchAppointments(token, 9783);
})
.then(function(results){
    //The API data
	console.log(results);
});

File: examples/csharp/net2.0/searchAppointments.cs
using System;
using System.Net;
using System.IO;

namespace ConsoleApplication
{
    class Program
    {
        static void Main(string[] args)
        {
            string baseUrl = "http://[SERVERNAME]/SRSAPI/Generic/";

            string username = "userName";
            string password = "password";

            string token = Request(baseUrl + "Token", null, String.Format("{{\"username\":\"{0}\",\"password\":\"{1}\",\"dataSourceId\":0}}", username, password));

            // Deserialize into an object
            string appointments = Request(baseUrl + "Appointment?PersonId=9783", token);
        }


        static string Request(string url, string authToken)
        {
            return Request(url, authToken, null);
        }

        static string Request(string url, string authToken, string data)
        {
            if (string.IsNullOrEmpty(url))
            {
                throw new ArgumentNullException("url");
            }

            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(new Uri(url));

            if (string.IsNullOrEmpty(data))
            {
                request.Method = "GET";
            }
            else
            {
                request.Method = "POST";
            }

            if (!string.IsNullOrEmpty(authToken))
            {
                request.Headers.Add("Authorization", authToken);
            }

            request.ContentType = "application/json";
            request.Accept = "application/json";

            if (!string.IsNullOrEmpty(data))
            {
                using(StreamWriter sw = new StreamWriter(request.GetRequestStream()))
                {
                    sw.Write(data);
                }
            }

            WebResponse response = request.GetResponse();

            using (StreamReader sr = new StreamReader(response.GetResponseStream()))
            {
                string responseData = sr.ReadToEnd();
                return responseData;
            }
        }

    }
}
